syntax = "proto3";

package skia_opt_metrics;

enum OptimizationStatus {
    UNKNOWN_STATUS = 0;
    SUCCESS = 1;
    FAILED = 2;
}

enum Optimization {
    UNKNOWN_OPTIMIZATION = 0;
    NO_OPT = 1;
    SKIA_RECORD_OPTS = 2;
    SKIA_RECORD_OPTS_2 = 3;
    SKI_PASS = 4;
}

message SkiPassMetrics {
    uint64 sk_record_ski_lang_duration_nano = 1;
    uint64 eqsat_duration_nano = 2;
    uint64 ski_lang_to_instructions_duration_nano = 3;
}

message OptimizationBenchmark {
    Optimization optimization_type = 1;
    OptimizationStatus optimization_status = 2;
    uint64 malloc_allocated_bytes = 3; // Valid only if optimization_status == SUCCESS
    uint64 optimization_duration_nano = 4;
    SkiPassMetrics ski_pass_metrics = 5;
}

message SkpBenchmark {
    string skp_name = 1;
    repeated OptimizationBenchmark optimization_benchmark_runs = 2;
}

message SkiaOptBenchmark {
    repeated SkpBenchmark skp_benchmark_runs = 1;
}
